analyses/interaction-plots/scripts/02-process_mutations.R

```diff
@@ -276,27 +277,53 @@ maf_filtered <- maf_df %>%
 gene_sample_counts <- maf_filtered %>%
   dplyr::filter(Entrez_Gene_Id > 0) %>% # remove unknowns
   dplyr::group_by(gene = Hugo_Symbol, sample = Tumor_Sample_Barcode) %>%
-  dplyr::tally(name = "mutations")
+  dplyr::tally(name = "mutations") %>%
+  dplyr::ungroup()
+
 
 # count # of samples mutated by gene
 gene_counts <- gene_sample_counts %>%
-  dplyr::filter(sample %in% samples) %>%
   dplyr::group_by(gene) %>%
   dplyr::summarize(
     mutant_samples = dplyr::n(),
     total_muts = sum(mutations),
     muts_per_sample = mean(mutations)
   ) %>%
+  dplyr::ungroup() %>%
   dplyr::arrange(
     desc(mutant_samples),
     desc(muts_per_sample)
   ) %>%
   dplyr::filter(mutant_samples >= opts$min_mutated |
     dplyr::row_number() <= 2) # keep at least 2 genes
 
+
+  
 # get most often mutated genes
 top_count_genes <- head(gene_counts, opts$max_genes)$gene
 
 cooccur_summary <- coocurrence(gene_sample_counts, top_count_genes)
 
 readr::write_tsv(cooccur_summary, out_file)
+
+# only count genes if requested
+if (is.na(disease_file)){
+  quit()
+}
+# count mutated samples by disease types
+gene_disease_counts <- gene_sample_counts %>%
+  dplyr::filter(gene %in% top_count_genes) %>%
+  dplyr::left_join(sample_meta, 
+                   by = c("sample" = "Kids_First_Biospecimen_ID")) %>%
+  dplyr::group_by(gene, disease = integrated_diagnosis) %>%
+  dplyr::summarize(mutant_samples = dplyr::n(),
+                   total_muts = sum(mutations),
+                   muts_per_sample = mean(mutations)) %>%
```

Comment:
I have not read far enough to know what this is used for, but making this ever so slightly more descriptive could be useful
```suggestion
                   avg_muts_per_sample = mean(mutations)) %>%
```

Reply:
I think it isn't used now, but it is available for use later.